### DISCUSSION TITLE
Limit on number of labels per image

### DISCUSSION THREAD

**Discussion by MyGHName** [ultralytics/yolov5#6577] [Q&A]
I'm working with large images having several thousand objects of interest. My detections results seem quite good during training on smaller images but the results on full size images seem to stop after labeling exactly 1000 objects. Is this a built in limit on the number of detected objects per image? Is there a way to change this?

Thank you.

**Comment by glenn-jocher** [comment]
@MyGHName see detect.py argparser for inference options:
https://github.com/ultralytics/yolov5/blob/f40854b61b2b347e1ff03a6fc0a765f1210a4fdc/detect.py#L224

**Reply by MyGHName** [reply]
Thank you for this. Unfort, I tried adding this code to my NB but received an error msg. See below. Any help would be appreciated.

Code:
!python parser.add_argument('--line-thickness', default=1, type=int, help='bounding box thickness (pixels)')
!python parser.add_argument('--max-det', type=int, default=5000, help='maximum detections per image') 
Errors:
/bin/bash: -c: line 0: syntax error near unexpected token `('
/bin/bash: -c: line 0: `python parser.add_argument('--line-thickness', default=1, type=int, help='bounding box thickness (pixels)')'
/bin/bash: -c: line 0: syntax error near unexpected token `('
/bin/bash: -c: line 0: `python parser.add_argument('--max-det', type=int, default=5000, help='maximum detections per image') '

**Reply by glenn-jocher** [reply]
@MyGHName oh, no that's not how argument work:

```
python detect.py --max-det 5000
```

**Reply by MyGHName** [reply]
Thanks, this worked. I owe you one!
